miner/worker.go

```diff
@@ -1032,6 +1045,9 @@ func (w *worker) collectPendingL1Messages() []types.L1MessageTx {
 		log.Crit("Failed to read last L1 message in L2 block", "l2BlockHash", w.chain.CurrentHeader().Hash())
 	}
 	startIndex := *nextQueueIndex
+	if (w.maxSkippedL1MsgIndex != nil) && (*w.maxSkippedL1MsgIndex >= startIndex) {
+		startIndex = *w.maxSkippedL1MsgIndex + 1
```

Comment:
This cannot deal with reorgs, `maxSkippedL1MsgIndex` should be assigned to the block.